test/gtest/soft_max_new.cpp

```diff
@@ -0,0 +1,161 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "test.hpp"
+#include <array>
+#include <cmath>
+#include <iostream>
+#include <iterator>
+#include <limits>
+#include <memory>
+#include <miopen/convolution.hpp>
+#include <miopen/miopen.h>
+#include <miopen/softmax.hpp>
+#include <miopen/tensor.hpp>
+#include <utility>
+#include <algorithm>
+
+#include "driver.hpp"
+#include "get_handle.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+
+#include <cstdint>
+#include <tuple>
+#include <gtest/gtest.h>
+
+struct InputDimension
+{
+    size_t N;
+    size_t C;
+    size_t H;
+    size_t W;
+
+    friend std::ostream& operator<<(std::ostream& os, const InputDimension& input)
+    {
+        return os << "(N: " << input.N << " C:" << input.C << " H:" << input.H << " W:" << input.W
+                  << ")";
+    }
+};
+// this is just try
+std::set<std::vector<int>> GetInputDimensions()
+{
+    return {
+        {1, 2, 3, 4},
+        {4, 3, 2, 1},
+    };
+}
+
+struct Scales
+{
+    float alpha;
+    float beta;
+
+    friend std::ostream& operator<<(std::ostream& os, const Scales& ab)
+    {
+        return os << "(alpha: " << ab.alpha << " beta:" << ab.beta << ")";
+    }
+};
+
+template <typename DataType>
+class SoftMaxTest
+    : public ::testing::TestWithParam<
+          std::tuple<InputDimension, miopenSoftmaxAlgorithm_t, miopenSoftmaxMode_t, Scales>>
+{
+public:
+    static std::vector<InputDimension> convertDim(const std::set<std::vector<int>>& input)
+    {
+
+        std::vector<InputDimension> result;
+        result.reserve(input.size());
+        std::transform(input.begin(),
+                       input.end(),
+                       std::back_inserter(result),
+                       [](const std::vector<int>& inp) {
+                           return InputDimension{static_cast<size_t>(inp[0]),
+                                                 static_cast<size_t>(inp[1]),
+                                                 static_cast<size_t>(inp[2]),
+                                                 static_cast<size_t>(inp[3])};
+                       });
+        return result;
+    }
+
+protected:
+    void SetUp() override
+    {
+        std::tie(input, algo, mode, scales) = GetParam();
+
+        auto&& handle = get_handle();
+
+        // initialze output tensor with same dim as first input tensor
+        output = tensor<DataType>{out_dim};
```

Comment:
```suggestion
        output = tensor<DataType>{input_dim};
```
See the first comment for the details.